src/gsCore/gsMath.h

```diff
@@ -321,6 +321,18 @@ template <typename T> int getSign(T val)
     return (T(0) < val) - (val < T(0));
 }
 
+/// Return smallest difference between two (integer) numbers as size_t
+/// especially the case abs_diff(INT32_MAX, INT32_MIN) := UINT32_MAX is correct
+template <typename T>
+inline size_t abs_diff(T a, T b)
```

Comment:
Why size_t?

It should be the unsigned type fitting to T (cf. https://en.cppreference.com/w/cpp/types/make_unsigned)

Reply:
THX, didn't know that.

Reply:
cool, that needs not the code to prevent abs_diff(>=0,INT32_MIN)

Reply:
its c++11 only. so, we need our own version

Reply:
like in tensortools

template<class T> struct make_unsigned {};
template<> struct<int> make_unsigned { typedef unsigned type; };
// ...

Reply:
> like in tensortools
> 
> template<class T> struct make_unsigned {};
> template<> struct make_unsigned<int> { typedef unsigned type; };
> template<> struct make_unsigned<unsigned> { typedef unsigned type; };
> // ...

can't find it in our code. But like [this](https://stackoverflow.com/questions/18183919/c-template-integer-types-adequate-unsigned-type)?